tracer/src/Datadog.Trace/PDBs/SourceLink/AzureDevOpsSourceLinkUrlParser.cs

```diff
@@ -64,6 +70,35 @@ internal override bool TryParseSourceLinkUrl(Uri uri, [NotNullWhen(true)] out st
         return false;
     }
 
+    private static string? BuildRepositoryUrl(Uri uri)
+    {
+        var segments = uri.AbsolutePath.Split(['/'], StringSplitOptions.RemoveEmptyEntries);
+        if (segments.Length < 5)
+        {
+            return null;
+        }
+
+        if (uri.Host.EndsWith("visualstudio.com", StringComparison.OrdinalIgnoreCase))
+        {
+            // Legacy format: https://{organization}.visualstudio.com
+            var project = segments[0];
+            var repoName = segments[4];
+            return $"https://{uri.Host}/{project}/_git/{repoName}";
+        }
+        else if (uri.Host.EndsWith("dev.azure.com", StringComparison.OrdinalIgnoreCase))
+        {
+            // New format: https://dev.azure.com/{organization}
+            var organization = segments[0];
+            var project = segments[1];
+            var repoName = segments[5];
+            return $"https://{uri.Host}/{organization}/{project}/_git/{repoName}";
+        }
+        else
+        {
+            throw new NotSupportedException($"Unsupported Azure DevOps host: {uri.Host}");
```

Comment:
Do we really want to throw an exception here? 🤔 Wouldn't it better to just log a warning and return null, seeing as we're going to catch it immediately in the calling code and log anyway - seems more expensive than necessary? 🤔

Reply:
Great point, fixed in 6f2037fa900ce7cedf7e54f65ac51dc3ad582b6e